xml_loader/shift_generation.py

```diff
@@ -143,70 +145,131 @@ def get_employee_lists(root):
         "employee_contracted_hours": employee_contracted_hours,
     }
 
+def get_demand_pairs(demand, day):
```

Comment:
Add a docstring comment to explain what this function does. Use """ YOUR COMMENT """.